spec.md

```diff
@@ -778,6 +800,182 @@ message Volume {
   // pre-populated with data from the specified source.
   // This field is OPTIONAL.
   VolumeContentSource content_source = 4;
+
+  // Specifies where (regions, zones, racks, etc.) the provisioned
+  // volume is accessible from.
+  // A plugin that returns this field must also set the
+  // ACCESSIBILITY_CONSTRAINTS plugin capability.
+  // An SP may specify multiple topologies to indicates the volume is
+  // accessible from multiple locations.
+  // COs may use this information along with the topology information
+  // returned by NodeGetId to ensure that a given volume is accessible
+  // from a given node when scheduling workloads.
+  // This field is OPTIONAL. If it is not specified, the CO MAY assume
+  // the volume is equally accessible from all nodes in the cluster and
+  // may schedule workloads referencing the volume on any available
+  // node.
+  //
+  // Example 1:
+  //   accessible_topology = {"region": "R1", "zone": "Z2"}
+  // Indicates a volume accessible only from the "region" "R1" and the
+  // "zone" "Z2".
+  //
+  // Example 2:
+  //   accessible_topology =
+  //     {"region": "R1", "zone": "Z2"},
+  //     {"region": "R1", "zone": "Z3"} 
+  // Indicates a volume accessible from both "zone" "Z2" and "zone" "Z3"
+  // in the "region" "R1".
+  repeated Topology accessible_topology = 5;
+}
+
+message TopologyRequirement {
```

Comment:
It seems to me that this should be presented in `GetCapacityRequest` as well, so the CO can query the capacity within a domain.

Reply:
Done